src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -749,8 +749,16 @@ uct_ib_mlx5_select_sl(const uct_ib_iface_config_t *ib_config,
     ucs_string_buffer_init(&sls_with_ar_str);
     ucs_string_buffer_init(&sls_without_ar_str);
 
-    if (((ar_enable == UCS_YES) || (ar_enable == UCS_TRY)) &&
-        (sls_with_ar != 0)) {
+    if (ar_enable == UCS_AUTO) {
+        /* selects SL requested by a user */
+        sl                    = ucs_ffs64(sl_allow_mask);
+        if (have_sl_mask_cap) {
```

Comment:
should it be vice versa? It is unknown if you do not have sl caps

Reply:
fixed